### PR TITLE
story/WOR-11/verification-du-numero-de-ticket-cote-api

### PR DESCRIPTION
**Pull request by remirubis**
## Description

I have added the entrance ticket management feature to Vivatech within the REST API. With this new feature, users can now purchase tickets for the event, and the system will manage the ticket's validity period, user information, and other details. It's just a mock of a potential implementation in vivatech api for internal usage.

## Changes

- Created a new User model.
- Set up the UserController to handle requests related to users.
- Added necessary packages and configured the environment for the user feature.
- Added routes for handling CRUD operations on users.
- Added tests to verify that all routes are working correctly.

### CODE REVIEW

**Review by Maengdok — DISMISSED** [review]

File: src/controllers/ticket.controllers.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    const { ticketId } = req.params;
+
+    if (!ticketId) {
+      errorFormatter(res, 400, ErrorMessages.MALFORMED_DATA);
+      return;
+    }
+
+    const ticket: ITicket | null = await Ticket.findOne({ ticketId }).select(['-__v', '-_id']);
+    if (!ticket) {
+      errorFormatter(res, 400, ErrorMessages.MALFORMED_DATA);
```
Comment (kilianpichard) [inline src/controllers/ticket.controllers.ts]: Not found instead ?

File: src/controllers/ticket.controllers.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    const { ticketId } = req.params;
+
+    if (!ticketId) {
+      errorFormatter(res, 400, ErrorMessages.MALFORMED_DATA);
+      return;
+    }
+
+    const ticket: ITicket | null = await Ticket.findOne({ ticketId }).select(['-__v', '-_id']).populate('user');
+    if (!ticket) {
+      errorFormatter(res, 400, ErrorMessages.MALFORMED_DATA);
```
Comment (kilianpichard) [inline src/controllers/ticket.controllers.ts]: Not found instead ?

File: src/models/ticket.model.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+ticketSchema.pre('save', async function (this: ITicket, next) {
+  if (!this.isNew) {
+    return next();
+  }
+
+  const ticketExists = await this.$model('Tickets').exists({ ticketId: this.ticketId });
+
+  if (ticketExists) {
+    // Generate another ticket number if it already exists
```
Comment (kilianpichard) [inline src/models/ticket.model.ts]: Are we generating ticket ? Or Vivatech does instead and we are just stocking it

**Review by kilianpichard — CHANGES_REQUESTED** [review]
Nice job ! ;)

File: src/models/ticket.model.ts (hunk 3 above)
Comment (remirubis) [reply src/models/ticket.model.ts]: Pour le bien du projet j'ai recréé un système de génération de ticket côté API 🤷🏼‍♂️

File: src/controllers/ticket.controllers.ts (hunk 2 above)
Comment (remirubis) [reply src/controllers/ticket.controllers.ts]: Good point, wrong var haha !

**Pull request closed without merging**